src/utils/graphql.js

```diff
@@ -4,6 +4,69 @@ import { HttpLink } from 'apollo-link-http'
 import { WebSocketLink } from 'apollo-link-ws'
 import { getMainDefinition } from 'apollo-utilities'
 import { InMemoryCache } from 'apollo-cache-inmemory'
+import { SubscriptionClient } from 'subscriptions-transport-ws'
+import store from '@/store'
+
+/**
+ * Create the HTTP and WebSocket URLs for an ApolloClient.
+ *
+ * @private
+ * @return {{wsUrl: string, httpUrl: string}}
+ * @private
+ */
+export function createGraphQLUrls () {
+  // TODO: revisit this and evaluate other ways to build the GraphQL URL - not safe to rely on window.location (?)
+  const baseUrl = `${window.location.hostname}${window.location.port ? ':' + window.location.port : ''}${window.location.pathname}`
+  const httpUrl = `${window.location.protocol}//${baseUrl}graphql`
+  const wsUrl = `${window.location.protocol.startsWith('https') ? 'wss' : 'ws'}://${baseUrl}subscriptions`
+  return {
+    httpUrl: httpUrl,
+    wsUrl: wsUrl
+  }
+}
+
+/**
+ * Create a subscription client.
+ *
+ * @private
+ * @param {string} wsUrl - WebSocket subscription URL
+ * @param {{
+ *   reconnect: boolean,
+ *   lazy: boolean
+ * }} options - SubscriptionClient options (only two main options added here, see their doc for more)
+ * @param {*} wsImpl
+ * @return {SubscriptionClient} a subscription client
+ */
+export function createSubscriptionClient (wsUrl, options = {}, wsImpl = null) {
+  const opts = Object.assign({
+    reconnect: true,
+    lazy: false
```

Comment:
We need these two default values. But callers (especially unit tests) are free to change. For unit tests, you want to have `lazy: true`, so that the `SubscriptionClient` doesn't try to create the connection, as node/mocha don't have a WebSocket implementation, resulting in runtime error.